rs/sns/testing/src/sns.rs

```diff
@@ -335,43 +319,74 @@ pub async fn upgrade_sns_controlled_test_canister<C: CallCanisters + ProgressNet
     let test_canister_wasm =
         Wasm::from_location_specified_by_env_var("sns_testing_canister", features).unwrap();
     let modified_test_canister_wasm = modify_wasm_bytes(&test_canister_wasm.bytes(), 42);
-    let temp_dir = TempDir::new().unwrap();
-    let new_wasm_path = temp_dir.path().join("new_test_canister.wasm");
-    {
-        let mut new_wasm_file = File::create(&new_wasm_path).unwrap();
-        new_wasm_file
-            .write_all(&modified_test_canister_wasm)
-            .unwrap();
-        new_wasm_file.flush().unwrap();
-    }
 
-    let icp = Tokens::from_tokens(10).unwrap();
-    convert_icp_to_cycles(dev_participant_agent, icp).await;
+    // TODO: @rvem: It's impossible to use 'upgrade_sns_controlled_canister::exec' function to upgrade the canister
+    // using the ic-agent backend on the network run by PocketIC because pocket-ic-server currently doesn't support
+    // calls to the management canister ('aaaaa-aa'), hence for now the upgrade is done using a single 'manage_neuron'
+    // call to the governance canister.
+    // let temp_dir = TempDir::new().unwrap();
+    // let new_wasm_path = temp_dir.path().join("new_test_canister.wasm");
+    // {
+    //     let mut new_wasm_file = File::create(&new_wasm_path).unwrap();
+    //     new_wasm_file
+    //         .write_all(&modified_test_canister_wasm)
+    //         .unwrap();
+    //     new_wasm_file.flush().unwrap();
+    // }
+
+    // let icp = Tokens::from_tokens(10).unwrap();
+    // convert_icp_to_cycles(dev_participant_agent, icp).await;
+
+    // let neuron_id = get_caller_neuron(dev_participant_agent, sns.governance)
+    //     .await
+    //     .unwrap();
+    // let candid_arg = (candid::IDLArgs {
+    //     args: vec![candid::IDLValue::try_from_candid_type(&upgrade_arg).unwrap()],
+    // })
+    // .to_string();
+    // let upgrade_args = UpgradeSnsControlledCanisterArgs {
+    //     sns_neuron_id: Some(ParsedSnsNeuron(neuron_id)),
+    //     target_canister_id: canister_id,
+    //     wasm_path: new_wasm_path,
+    //     candid_arg: Some(candid_arg),
+    //     proposal_url: Url::try_from("https://github.com/dfinity/ic").unwrap(),
+    //     summary: "Upgrade Test canister".to_string(),
+    // };
+    // let UpgradeSnsControlledCanisterInfo { proposal_id, .. } =
+    //     upgrade_sns_controlled_canister::exec(upgrade_args, dev_participant_agent)
+    //         .await
+    //         .expect("Failed to upgrade the canister");
+    // let proposal_id = proposal_id.unwrap();
```

Comment:
Is this intentionally left commented?

Reply:
Yep, this part of the commented code refers to the TODO above